packages/core/rxjs-interop/test/take_until_destroyed_spec.ts

```diff
@@ -63,4 +80,16 @@ describe('takeUntilDestroyed', () => {
     source$.next(2);
     expect(last).toBe(1);
   });
+
+  it('should unregister listener if observable is unsubscribed', () => {
+    const destroyRefMock = new DestroyRefMock();
+
+    const subscription = new BehaviorSubject(0).pipe(takeUntilDestroyed(destroyRefMock)).subscribe()
+
+    expect(destroyRefMock.handles.length).toBe(1);
+
+    subscription.unsubscribe()
+
+    expect(destroyRefMock.handles.length).toBe(0);
```

Comment:
```suggestion
    const destroyRef = TestBed.inject(DestroyRef);
    const unregisterFn = jasmine.createSpy();
    spyOn(destroyRef, 'onDestroy').and.returnValue(unregisterFn);

    const subscription = new BehaviorSubject(0).pipe(takeUntilDestroyed(destroyRef)).subscribe();
    subscription.unsubscribe();

    expect(unregisterFn).toHaveBeenCalled();
```

The mock is a little heavy-weight for what's really needed and is testing a lot of the mock's implementation. All we really need to know is that the unregister function is called.